Subject: [PATCH] Prevent destroy surface during window replacement.

When a window replacement request is ongoing, the old window shouldn't destroy it's surface
until window replacement finished.

Step:
 1. Set Transition animation scale as (.5X)
 2. Go to Clock, enter the Clock's settings
 3. Click Silence after, choose another times.
 4. Click Exit Button.

diff --git a/services/core/java/com/android/server/wm/WindowStateAnimator.java b/services/core/java/com/android/server/wm/WindowStateAnimator.java
index afd451d..ec7c554 100644
--- a/services/core/java/com/android/server/wm/WindowStateAnimator.java
+++ b/services/core/java/com/android/server/wm/WindowStateAnimator.java
@@ -583,6 +583,10 @@
             return;
         }
 
+        if (mWin.mReplacingRemoveRequested) {
+            return;
+        }
+
         if (WindowManagerService.localLOGV || DEBUG_ADD_REMOVE) Slog.v(TAG,
                 "Exit animation finished in " + this + ": remove=" + mWin.mRemoveOnExit);
 
